rita_common/src/payment_controller/mod.rs

```diff
@@ -143,10 +156,176 @@ pub async fn tick_payment_controller() {
 
 /// This is called by debt_keeper to make payments. It sends a
 /// PaymentTx to the `mesh_ip` in its `to` field.
-async fn make_payment(mut pmt: PaymentTx) -> Result<(), PaymentControllerError> {
+async fn make_payment(pmt: PaymentTx) -> Result<(), PaymentControllerError> {
     let common = settings::get_rita_common();
     let network_settings = common.network;
     let payment_settings = common.payment;
+    let system_chain = payment_settings.system_chain;
+
+    match system_chain {
+        SystemChain::Althea => make_althea_payment(pmt, payment_settings, network_settings).await,
+        SystemChain::Xdai => make_xdai_payment(pmt, payment_settings, network_settings).await,
+        SystemChain::Rinkeby => {
+            warn!("Payments on Rinkeby not currently supported!");
+            Ok(())
+        }
+        SystemChain::Ethereum => {
+            warn!("Payments on Ethereum not currently supported!");
+            Ok(())
+        }
+    }
+}
+
+async fn make_althea_payment(
+    mut pmt: PaymentTx,
+    payment_settings: PaymentSettings,
```

Comment:
while it is true that we can know right away if our payment succeeded or failed we need to be able to query the payment hash to know if we where disconnected before a payment went through but it actually did go through. You think this wouldn't happen often but it def happens often enough.

Reply:
Sending pmt to 'validate_later()' when we receive an Ok() from 'send_microtx', and am returning an Err() when I receive an Err() from 'send_microtx'. I would either need confirmated that receving an Err means tx failed, or a way to get a tx_hash back to send to validate_later